edge-modules/MetricsCollector/IoTHubMetricsSync.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+namespace MetricsCollector
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    public class IoTHubMetricsSync : IMetricsSync
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("MetricsCollector");
+        readonly MessageFormatter messageFormatter;
+        readonly Scraper scraper;
+        readonly ModuleClient moduleClient;
+
+        public IoTHubMetricsSync(MessageFormatter messageFormatter, Scraper scraper, ModuleClient moduleClient)
+        {
+            this.messageFormatter = Preconditions.CheckNotNull(messageFormatter);
+            this.scraper = Preconditions.CheckNotNull(scraper);
+            this.moduleClient = Preconditions.CheckNotNull(moduleClient);
+        }
+
+        public async Task ScrapeAndSync()
+        {
+            try
+            {
+                IEnumerable<string> scrapedMetrics = await this.scraper.Scrape();
+                IList<Message> messages =
+                        scrapedMetrics.SelectMany(prometheusMetrics => this.messageFormatter.Build(prometheusMetrics)).ToList();
+                await this.moduleClient.SendEventBatchAsync(messages);
+                Logger.LogInformation($"Sent metrics as {messages.Count} messages");
+            }
+            catch (Exception e)
+            {
+                Logger.LogError($"Error scraping and syncing metrics - {e}");
```

Comment:
It looks like `SendEventBatchAsync()` does not retry the upload if the message failed. 
Should we do a retry here for a not-so-critical http (503, 429, 301) here? 
See Mozilla standard: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After

Reply:
Can edgeHub return a 503? I'm pretty sure it wouldn't return a 429 or 301.